### ISSUE TITLE
[maple] SurfaceFlinger cannot load libsdmextension.so

### ISSUE THREAD

**Issue by jamuir** [sonyxperiadev/bug_tracker#55]
**Platform:** yoshino
**Device:** maple
**Kernel version:** 4.4
**Android version:** 7.1.1

**Description**
On boot, SurfaceFlinger starts up and tries to load `libsdmextension.so`.
It fails due to an unresolved symbol:

```
01-12 03:01:59.604 I/SurfaceFlinger(  614): SurfaceFlinger is starting
01-12 03:01:59.604 I/SurfaceFlinger(  614): SurfaceFlinger's main thread ready to run. Initializing graphics H/W...
01-12 03:01:59.604 E/cutils-trace(  614): Error opening trace file: No such file or directory (2)
01-12 03:01:59.608 D/libEGL  (  614): loaded /vendor/lib64/egl/libEGL_adreno.so
01-12 03:01:59.658 D/libEGL  (  614): loaded /vendor/lib64/egl/libGLESv1_CM_adreno.so
01-12 03:01:59.667 D/libEGL  (  614): loaded /vendor/lib64/egl/libGLESv2_adreno.so
01-12 03:01:59.685 I/Adreno  (  614): QUALCOMM build                   : 8209866, I528a81912f
01-12 03:01:59.685 I/Adreno  (  614): Build Date                       : 04/01/17
01-12 03:01:59.685 I/Adreno  (  614): OpenGL ES Shader Compiler Version: XE031.14.00.02
01-12 03:01:59.685 I/Adreno  (  614): Local Branch                     : 
01-12 03:01:59.685 I/Adreno  (  614): Remote Branch                    : 
01-12 03:01:59.685 I/Adreno  (  614): Remote Branch                    : 
01-12 03:01:59.685 I/Adreno  (  614): Reconstruct Branch               : 
01-12 03:01:59.747 I/Adreno  (  614): PFP: 0x005ff087, ME: 0x005ff063
01-12 03:01:59.759 I/SurfaceFlinger(  614): EGL information:
01-12 03:01:59.759 I/SurfaceFlinger(  614): vendor    : Android
01-12 03:01:59.759 I/SurfaceFlinger(  614): version   : 1.4 Android META-EGL
01-12 03:01:59.759 I/SurfaceFlinger(  614): extensions: EGL_KHR_get_all_proc_addresses EGL_ANDROID_presentation_time EGL_KHR_swap_buffers_with_damage EGL_ANDROID_create_native_client_buffer EGL_ANDROID_front_buffer_auto_refresh EGL_KHR_image EGL_KHR_image_base EGL_KHR_lock_surface EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_3D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_reusable_sync EGL_KHR_fence_sync EGL_KHR_create_context EGL_KHR_surfaceless_context EGL_EXT_create_context_robustness EGL_ANDROID_image_native_buffer EGL_KHR_wait_sync EGL_ANDROID_recordable EGL_KHR_partial_update EGL_KHR_create_context_no_error EGL_KHR_mutable_render_buffer EGL_EXT_yuv_surface EGL_EXT_protected_content 
01-12 03:01:59.759 I/SurfaceFlinger(  614): Client API: OpenGL_ES
01-12 03:01:59.759 I/SurfaceFlinger(  614): EGLSurface: 8-8-8-8, config=0x7f25586420
01-12 03:01:59.763 I/SurfaceFlinger(  614): OpenGL ES informations:
01-12 03:01:59.763 I/SurfaceFlinger(  614): vendor    : Qualcomm
01-12 03:01:59.763 I/SurfaceFlinger(  614): renderer  : Adreno (TM) 540
01-12 03:01:59.763 I/SurfaceFlinger(  614): version   : OpenGL ES 3.2 V@207.0 (GIT@8209866, I528a81912f) (Date:04/01/17)
01-12 03:01:59.763 I/SurfaceFlinger(  614): extensions: GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_sync GL_OES_vertex_half_float GL_OES_framebuffer_object GL_OES_rgb8_rgba8 GL_OES_compressed_ETC1_RGB8_texture GL_AMD_compressed_ATC_texture GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_hdr GL_OES_texture_compression_astc GL_OES_texture_npot GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_OES_texture_3D GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_QCOM_alpha_test GL_OES_depth24 GL_OES_packed_depth_stencil GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_EXT_sRGB GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_EXT_texture_type_2_10_10_10_REV GL_EXT_texture_sRGB_decode GL_OES_element_index_uint GL_EXT_copy_image GL_EXT_geometry_shader GL_EXT_tessellation_shader GL_OES_texture_stencil8 GL_EXT_shader_io_blocks GL_OES_shader_image_atomic GL_OES_sample_variables GL_EXT_texture_border_clamp GL_EXT_multisampled_render_to_texture
01-12 03:01:59.763 I/SurfaceFlinger(  614): GL_MAX_TEXTURE_SIZE = 16384
01-12 03:01:59.763 I/SurfaceFlinger(  614): GL_MAX_VIEWPORT_DIMS = 16384
01-12 03:01:59.773 W/SDM     (  614): CoreImpl::Init: Unable to load = libsdmextension.so, error = dlopen failed: cannot locate symbol "_ZN3sdm11TransformHVERKNS_9LayerRectES2_RKNS_14LayerTransformEPS0_" referenced by "/system/vendor/lib64/libsdmextension.so"...
```

**Symptoms**

**How to reproduce**
reboot your device and capture logcat when after the ui comes up.

**Comment by jerpelea** [comment]
will be fixed in v7

**Comment by jamuir** [comment]
thanks.  did you mean v6?  the SW package for yoshino is currently v5.

**Comment by notz** [comment]
Look here: https://developer.sony.com/develop/open-devices/latest-updates

**Comment by jamuir** [comment]
Thanks for the pointer.  Quoting from that web page:

> 2017-11-03: Update of AOSP Nougat binaries (Android 7.1 – Kernel 4.4) for Xperia X and Xperia X Compact, Xperia XZ Premium, Xperia XZ1 and Xperia XZ1 Compact, Xperia XZ, Xperia XZs and Xperia X Performance. Platform fixes for all devices. Camera fix for Xperia X and Xperia X Compact. Connectivity fix for Xperia XZ Premium, XZ1 and XZ1 Compact.

> 2017-10-23: Update of AOSP Nougat binaries (Android 7.1 – Kernel 3.10) for Xperia XZ Premium, Xperia XZ1 and Xperia XZ1 Compact. Added version property, updated the firmware and fixed the light sensor for all devices. Added support for Xperia XZ1 and Xperia XZ1 Compact. Added secondary camera for Xperia XZ1 Compact.

I am only given the option to download v5 (`98f513dc6d60`, 23-Oct) from the sony web page, but it looks like a v6 package was prepared on 3-Nov.

@jerpelea is the web page up to date?

https://developer.sony.com/file/download/software-binaries-for-aosp-nougat-android-7-1-kernel-4-4-yoshino/
https://developer.sonymobile.com/downloads/software-binaries/software-binaries-for-aosp-nougat-android-7-1-kernel-4-4-yoshino/

**Comment by jerpelea** [comment]
v6 is not released yet for 7.1
